tests/status/worktree.c

```diff
@@ -1280,3 +1280,32 @@ void test_status_worktree__with_directory_in_pathlist(void)
 	git_status_list_free(statuslist);
 }
 
+void test_status_worktree__at_head_parent(void)
+{
+  git_repository *repo = cl_git_sandbox_init("empty_standard_repo");
+  git_status_options opts = GIT_STATUS_OPTIONS_INIT;
+  git_status_list *statuslist;
+  git_tree *parentTree;
+  const git_status_entry *status;
+  
+  cl_git_mkfile("empty_standard_repo/file1", "ping");
+  stage_and_commit(repo, "file1");
+  
+  git_repository_head_tree(&parentTree, repo);
+  
+  cl_git_mkfile("empty_standard_repo/file2", "pong");
+  stage_and_commit(repo, "file2");
+
+  cl_git_rewritefile("empty_standard_repo/file2", "pyng");
+  
+  opts.show = GIT_STATUS_SHOW_INDEX_AND_WORKDIR;
+  git_status_list_new_at(&statuslist, repo, &opts, parentTree);
```

Comment:
Please wrap this in an cl_assert, so that we do not segfault on a NULL pointer here.